src/jit/morph.cpp

```diff
@@ -7060,7 +7060,13 @@ void Compiler::fgMorphTailCall(GenTreeCall* call)
         }
 #endif // _TARGET_X86_
 
+#if !defined(_TARGET_X86_)
         if (call->NeedsNullCheck())
+#else
+        // When targeting x86, we always need to perform a null check on the `this` argument before tail calling to a
+        // virtual dispatch stub.
```

Comment:
I believe that this is is because of where we want to A/V to occur.  A/V are alway properly mapped into Null Reference exception when the occur in managed code. 

If they occur in unmanaged code then a lot depends upon what kind of special handling has been done in the VM.  For common stuff like normal Virtual Stub Dispatch we have  code in the VM to properly trap and change the A/V into a NRE, but probably for the  tail call variant of VSD  we don't have it. (and whomever looked at it decided it was tricky / too much work to get it right)